frontend/src/components/module-viewer/ModuleViewer.tsx

```diff
@@ -155,12 +160,41 @@ const ModuleViewer = ({
   }, [courseID, moduleIndex]);
 
   useEffect(() => {
-    if (courseData) {
-      // Save to context
-      getLessons({
-        variables: { ids: courseData.course.modules[moduleIndex].lessons },
-      });
-    }
+    const fetchLessons = async () => {
+      const module = courseData?.course.modules[moduleIndex];
+      if (courseData) {
+        // Save to context
+        const { data: lessonRes } = await getLessons({
+          variables: { ids: module.lessons },
+        });
+        const { data: lessonProgressRes } = await getProgressData({
+          variables: {
+            userId: authenticatedUser?.id,
+            lessonIds: lessonRes.lessons.map(
+              (lesson: LessonResponse) => lesson?.id,
+            ),
+          },
+        });
+        const progresses = lessonProgressRes?.lessonProgress || [];
+        const lessons = new Set(
```

Comment:
Instead of casting, does Set have a typed ctor? I.e. `new Set<string>()`

Reply:
Also it would be good to rename this to be clearer, something like `lessonsWithProgress`

Reply:
renamed to completedLessonIds

Reply:
renamed to completedLessons since now filtering by if `completedAt` exists